File: utilslibrary/src/main/java/com/sincerity/utilslibrary/http/HttpCallBack.java
package com.sincerity.utilslibrary.http;

/**
 * Created by Sincerity on 2019/9/3.
 * 描述：成功失败的回调
 */
public abstract class HttpCallBack<T> {
    /**
     * 成功回调
     *
     * @param resultJson 返回的Json数据
     * @param result     实体类
     */
    public abstract void onSuccess(String resultJson, T result);

    /**
     * 失败回调
     *
     * @param ex 异常
     */
    public abstract void onFail(Throwable ex);
}

File: framelibrary/src/main/java/com/sincerity/framelibrary/base/HttpCallBack.kt
package com.sincerity.framelibrary.base

import android.content.Context
import com.google.gson.Gson
import com.sincerity.utilslibrary.httputils.CallBack
import com.sincerity.utilslibrary.httputils.HttpUtils

/**
 * Created by Sincerity on 2019/11/20.
 * 描述： 遗留问题 1. 处理后台返回的异常数据信息 用OkHttp的拦截器去做
 */
abstract class HttpCallBack<T> : CallBack {
    override fun onPreExecute(context: Context, params: MutableMap<String, Any>?) {
        params?.put("", "")
        onPreExecute()
    }

    //开始执行
    fun onPreExecute() {
    }

    override fun onSuccess(result: String) {
        val gson = Gson()
        val objResult: T = gson.fromJson(result, HttpUtils.Companion.analysisClazzInfo(this)) as T
        onSuccess(objResult)
    }

    abstract fun onSuccess(result: T)

}
